### PR TITLE
Properties

### PR DESCRIPTION
**Pull request by leod**
- Add tag properties
- Rename CanBeDamaged to Character

### CODE REVIEW

File: Brimstone/EntityTypes/Character.cs [hunk 1]
```diff
@@ -2,10 +2,10 @@
 
 namespace Brimstone
 {
-	public abstract partial class CanBeDamaged : Entity
+	public abstract partial class Character : Entity
```
Comment (djkaty) [inline Brimstone/EntityTypes/Character.cs]: Can you redo it as an interface ICharacter with Hit as an extension method please :)

File: Brimstone/EntityTypes/Game.cs [hunk 2]
```diff
@@ -23,6 +23,9 @@ public class FuzzyGameComparer : IEqualityComparer<Game>
 
 	public partial class Game : Entity, IZoneOwner, IFormattable
 	{
+		// Game settings
+		public int MaxNumMinionsOnBoard { get; private set; } = 7;
```
Comment (djkaty) [inline Brimstone/EntityTypes/Game.cs]: Rename to MaxMinionsOnBoard

File: Brimstone/TagProperties.cs [hunk 3]
```diff
@@ -69,6 +89,16 @@ public partial class Player : Entity, IZoneOwner
 			set { this[GameTag.PLAYSTATE] = (int)value; }
 		}
 
+		public int Mana {
```
Comment (djkaty) [inline Brimstone/TagProperties.cs]: Hmm.... I wonder if this should be BaseManaThisTurn or something, we're going to need like ManaAvailable, ManaUsed, ExtraMana and all kinds of crap

File: Brimstone/EntityTypes/Character.cs (hunk 1 above)
Comment (leod) [inline Brimstone/EntityTypes/Character.cs]: What about the tag properties for Character?

File: Brimstone/TagProperties.cs [hunk 4]
```diff
@@ -104,9 +143,81 @@ public abstract partial class CanBeDamaged : Entity
 			}
 		}
 
-		public int Damage {
-			get { return this[GameTag.DAMAGE]; }
-			set { this[GameTag.DAMAGE] = value; }
+		public int IncomingDamageMultiplier {
```
Comment (djkaty) [inline Brimstone/TagProperties.cs]: So it seems like we don't have clarity on whether this tag exists anymore so leave it out for now. I have not yet updated the enums to match Hearthstone 6.

File: Brimstone/TagProperties.cs [hunk 5]
```diff
@@ -104,9 +143,81 @@ public abstract partial class CanBeDamaged : Entity
...
+
+	public partial class Minion : Character, IMinion
+	{
+		public bool CantBeTargetedByAbilities {
+			get { return this[GameTag.CANT_BE_TARGETED_BY_ABILITIES] == 1; }
+			set { this[GameTag.CANT_BE_TARGETED_BY_ABILITIES ] = value ? 1 : 0; }
+		}
+
+		public bool HasBattlecry {
+			get { return this[GameTag.BATTLECRY] == 1; }
```
Comment (djkaty) [inline Brimstone/TagProperties.cs]: What's the use case for this? `entity.Card.Behaviours.Battlecry != null` will tell you if the card has a battlecry _that is implemented by Brimstone_. How are you using the tag?

File: Brimstone/EntityTypes/Character.cs (hunk 1 above)
Comment (djkaty) [inline Brimstone/EntityTypes/Character.cs]: Ugh, right ok, as you were :)

**Pull request merged**